bucket/qq.json

```diff
@@ -0,0 +1,59 @@
+{
+    "version": "9.7.3.28946",
+    "description": "An instant messaging tool that gives you the best way to keep in touch with your friends and family.",
+    "homepage": "https://im.qq.com",
+    "license": {
+        "identifier": "Proprietary",
+        "url": "https://ti.qq.com/agreement/index.html"
+    },
+    "url": "https://webcdn.m.qq.com/spcmgr/download/QQ9.7.3.28946.exe#/dl.7z",
+    "hash": "6d11c322a7deb69634de54ba3006f908fa53a09a63585aaf40b02d3dce6fd825",
+    "shortcuts": [
+        [
+            "Bin\\QQScLauncher.exe",
+            "QQ"
+        ]
+    ],
+    "installer": {
+        "script": [
+            "Get-Item \"$dir\\*\" -Exclude .rsrc | Remove-Item",
+            "Expand-7zipArchive \"$dir\\.rsrc\\2052\\MSI\\101\" \"Files\" -Destination \"$dir\\\" -Removal",
+            "Move-Item \"$dir\\Files\\*\" \"$dir\"",
+            "Expand-7zipArchive \"$dir\\.rsrc\\2052\\MSI\\102\" \"vcruntime140.dll\" -Destination \"$dir\\Bin\\\"",
+            "Expand-7zipArchive \"$dir\\.rsrc\\2052\\MSI\\102\" \"msvcp140.dll\" -Destination \"$dir\\Bin\\\"",
+            "Expand-7zipArchive \"$dir\\.rsrc\\2052\\MSI\\102\" \"msvcr100.dll\" -Destination \"$dir\\Bin\\\"",
+            "Expand-7zipArchive \"$dir\\.rsrc\\2052\\MSI\\102\" \"vcomp140.dll\" -Destination \"$dir\\Bin\\\"",
+            "Expand-7zipArchive \"$dir\\.rsrc\\2052\\MSI\\102\" \"msvcp100.dll\" -Destination \"$dir\\Bin\\\"",
+            "Expand-7zipArchive \"$dir\\.rsrc\\2052\\MSI\\102\" \"TXSSO\\TXSSOSetup.exe\" -Destination \"$dir\" -Removal",
+            "Expand-7zipArchive \"$dir\\TXSSO\\TXSSOSetup.exe\" \"TXSSO\" -Destination \"$dir\\bin\" -Removal"
+        ]
+    },
+    "post_install": [
+        "Echo \"$dir\\Files\" \"$dir\\.rsrc\" \"$dir\\TXSSO\" | Remove-Item -Recurse",
+        "if (!(Test-Path \"$persist_dir\")) {",
+        "    ensure \"$persist_dir\" | Out-Null",
+        "}",
+        "$TencentFilesFolder = [Environment]::GetFolderPath(\"MyDocuments\") + \"\\Tencent Files\" ",
+        "if (Test-Path $TencentFilesFolder) {",
+        "    Move-Item \"$TencentFilesFolder\\*\" \"$persist_dir\" ",
+        "    Remove-Item \"$TencentFilesFolder\" -Force }",
+        "New-Item \"$TencentFilesFolder\" -ItemType Junction -Target \"$persist_dir\" | Out-Null",
+        "$QQRoaming = \"$env:APPDATA\\Tencent\\QQ\" ",
+        "if (Test-Path $QQRoaming) {",
+        "    Move-Item \"$QQRoaming\\*\" \"$persist_dir\\QQRoaming\" ",
+        "    Remove-Item \"$QQRoaming\" -Force }",
+        "if (!(Test-Path \"$persist_dir\\QQRoaming\")) { New-Item \"$persist_dir\\QQRoaming\" -ItemType \"directory\" | Out-Null}",
+        "New-Item \"$QQRoaming\" -ItemType Junction -Target \"$persist_dir\\QQRoaming\" | Out-Null"
+    ],
+
+    "uninstaller": {
+        "script": "if ($purge) { Remove-Item \"$env:APPDATA\\Tencent\\QQ\" -Recurse -Force -ErrorAction SilentlyContinue }"
+    },
```

Comment:
Thoughts? /cc @ScoopInstaller/maintainers

Reply:
From the `post_install` I can see we're trying to _force_ persistence here. I do not advocate this: it can lead to all kinds of unexpected behaviour. If the app stores data in the Documents or the AppData folder, we should let it. 

Only if an application is _by design_ portable, we should manipulate its data.

Reply:
You're right!